bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/internal/events/ThreadedEventHandler.java

```diff
@@ -65,7 +65,7 @@ public ThreadedEventHandler(final Map<String, EventFactory> typedEventFactories,
                     final Event event = queue.poll(1, TimeUnit.HOURS);
                     if (event == null) {
                         logger.debug("Hey, you have really very few events.");
-                    } else if (event == notifyEvent) {
+                    } else if (event.equals(notifyEvent)) {
```

Comment:
Nope - we have a de-facto singleton instance of notifyEvent and are comparing it against events of potentially unknown sources where we don't know what they will do in their `equals()`-implementations, hence I'd suggest leaving this as it is.